Update README.md (#1)
franciscodua/codacy-test: README.md

> # codacy-test

Reviewer: Hello
```suggestion
Hello
```

> # codacy-test

Reviewer: Hello
```suggestion
Hello
world
```

> # codacy-test

Reviewer: Hello
```suggestion
Hello
world
```

> # codacy-test

Reviewer: Hello

> # codacy-test

Reviewer: Hello
```suggestion
Hello
world

123```

> # codacy-test

Reviewer: Hello
```suggestion
Hello
world

123
```

> # codacy-test

Reviewer: Hello
```suggestion
Hello
world

123
hello
```

> # codacy-test

Reviewer: Hello
```suggestion
Hello
world
```

> # codacy-test

Reviewer: Hello

> # codacy-test

Reviewer: Hello

> # codacy-test

Reviewer: Hello

> # codacy-test

Reviewer: Hello
```suggestion
Hello
world
```

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 0; Below](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

### Among Codacy’s features
```

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 0; Below](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

### Among Codacy’s features
```

> 

Reviewer: Hello
```suggestion
Hello
world
```

> 

Reviewer: Hello
```suggestion
Hello
world
```

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 0; Below](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

### Among Codacy’s features
```

> 

Reviewer: Hello

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 0; Below](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

### Among Codacy’s features
```

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 0; Below](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

### Among Codacy’s features
```

> ### Among Codacy’s features

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Identify new Static Analysis issues
```

> ### Among Codacy’s features

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Identify new Static Analysis issues
```

> ### Among Codacy’s features

Reviewer: Hello
```suggestion
Hello
world
```

> ### Among Codacy’s features

Reviewer: Hello

> ### Among Codacy’s features

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Identify new Static Analysis issues
```

> ### Among Codacy’s features

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Identify new Static Analysis issues
```

> ### Among Codacy’s features

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 0; Below](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

### Among Codacy’s features
```

> ### Among Codacy’s features

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 0; Below](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
### Among Codacy’s features

```

> ### Among Codacy’s features

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 0; Below](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
### Among Codacy’s features

```

> ### Among Codacy’s features

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Identify new Static Analysis issues
```

> - Identify new Static Analysis issues

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
- Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> - Identify new Static Analysis issues

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

* Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> - Identify new Static Analysis issues

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
- Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> - Identify new Static Analysis issues

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

* Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> - Identify new Static Analysis issues

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
- Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> - Identify new Static Analysis issues

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

* Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> - Identify new Static Analysis issues

Reviewer: Hello WORLD
```suggestion
```

> - Identify new Static Analysis issues

Reviewer: Hello WORLD
```suggestion
```

> - Identify new Static Analysis issues

Reviewer: Hello WORLD 2
```suggestion
```

> - Identify new Static Analysis issues

Reviewer: Hello WORLD 2
```suggestion
```

> - Identify new Static Analysis issues

Reviewer: Hello WORLD 2
```suggestion
```

> - Identify new Static Analysis issues

Reviewer: Hello WORLD 2
```suggestion
```

> - Identify new Static Analysis issues

Reviewer: Hello WORLD 2
```suggestion
```

> - Identify new Static Analysis issues

Reviewer: Hello WORLD 2
```suggestion
```

> - Identify new Static Analysis issues

Reviewer: Hello WORLD 2
```suggestion
```

> - Identify new Static Analysis issues

Reviewer: Playing with Lopes

> - Identify new Static Analysis issues

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
- Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> - Identify new Static Analysis issues

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

* Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> - Identify new Static Analysis issues

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Identify new Static Analysis issues
```

> - Identify new Static Analysis issues

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Identify new Static Analysis issues
```

> - Identify new Static Analysis issues

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Identify new Static Analysis issues
```

> - Identify new Static Analysis issues

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
- Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> - Identify new Static Analysis issues

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

* Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> * Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Auto-comments on Commits and Pull Requests
```

> * Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Auto-comments on Commits and Pull Requests
```

> * Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Auto-comments on Commits and Pull Requests
```

> * Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Auto-comments on Commits and Pull Requests
```

> * Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

* Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> * Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
- Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> * Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

* Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> * Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
- Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> * Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

* Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> * Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
- Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)
```

> * Commit and Pull Request Analysis with GitHub, BitBucket/Stash, GitLab (and also direct git repositories)

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Auto-comments on Commits and Pull Requests
```

> - Auto-comments on Commits and Pull Requests

Author: ```suggestion

- Auto-comments on Commits and Pull Requests
```

> - Auto-comments on Commits and Pull Requests

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Auto-comments on Commits and Pull Requests
```

> - Auto-comments on Commits and Pull Requests

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Auto-comments on Commits and Pull Requests
```

> - Auto-comments on Commits and Pull Requests

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Auto-comments on Commits and Pull Requests
```

> - Integrations with Slack, HipChat, Jira, YouTrack

Reviewer: Hello
```suggestion
Hello
world
```

- Track issues in Code Style, Security, Error Proneness, Performance, Unused Code and other categories

> Codacy also helps keep track of Code Coverage, Code Duplication, and Code Complexity.

Reviewer: Hello

> Codacy also helps keep track of Code Coverage, Code Duplication, and Code Complexity.

Reviewer: Some comment with line
```suggestion
Better this
```

## Codacy ESLint

# Adding new plugins / configs

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
1. Install the package using npm
```

> 1.  Install the package using npm

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
2. \[Plugins only\] Add the plugin to the plugins section in the file `src/eslintDefaultOptions.ts`
```

> 2.  \[Plugins only\] Add the plugin to the plugins section in the file `src/eslintDefaultOptions.ts`

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
3. \[Plugins only\] If the plugin has descriptions for rules on GitHub, reference them
```

3.  \[Plugins only\] If the plugin has descriptions for rules on GitHub, reference them
at `src/docGeneratorMain.ts` to include them on the generated documentation. To do this, add a section similar to the following example:

    ```typescript
    console.log("Generate xss description files")
    await docGenerator.downloadDocs(
      (pattern) =>
        `${githubBaseUrl}/Rantanen/eslint-plugin-xss/master/docs/rules/${pattern}.md`,
      "xss"
    )
    ```

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
4. Generate documentation so it adds the new plugin documentation
```

4.  Generate documentation so it adds the new plugin documentation

## Generating documentation

```bash
npm install
npm run generateDocs
```

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 0; Below](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

## Test changes to codacy-seed locally
```

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 0; Below](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

## Test changes to codacy-seed locally
```

> ## Test changes to codacy-seed locally

Author: Codacy has a fix for the issue: [(codacy-engine-typescript-seed)[https://github.com/codacy/codacy-engine-typescript-seed]](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
You may need to test changes that comes from our [codacy-engine-typescript-seed](https://github.com/codacy/codacy-engine-typescript-seed).
```

You may need to test changes that comes from our (codacy-engine-typescript-seed)[https://github.com/codacy/codacy-engine-typescript-seed].

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
1. Create a package with your changes on the seed:
```

1.  Create a package with your changes on the seed:
    - Don't forget to update the dependencies: `npm install`
    - Compile the library: `npm run compile`
    - Package the library: `npm pack`
        > This should generate a codacy-seed-0.0.1.tgz on your codacy-seed repository

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
2. Copy the `codacy-seed-0.0.1.tgz` into the root of this repository
```

2.  Copy the `codacy-seed-0.0.1.tgz` into the root of this repository

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
3. Install the package: `npm install codacy-seed-0.0.1.tgz`
```

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
3. Install the package: `npm install codacy-seed-0.0.1.tgz`
```

3.  Install the package: `npm install codacy-seed-0.0.1.tgz`

> 4.  Update Dockerfile and `.dockerignore` so you copy the `codacy-seed-0.0.1.tgz` inside the docker you will be building

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
    * Add `!codacy-seed-0.0.1.tgz` to your `.dockerignore`
```

> 4.  Update Dockerfile and `.dockerignore` so you copy the `codacy-seed-0.0.1.tgz` inside the docker you will be building

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
    -  Add `!codacy-seed-0.0.1.tgz` to your `.dockerignore`
```

> 4.  Update Dockerfile and `.dockerignore` so you copy the `codacy-seed-0.0.1.tgz` inside the docker you will be building

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
    -  Add `!codacy-seed-0.0.1.tgz` to your `.dockerignore`
```

> 4.  Update Dockerfile and `.dockerignore` so you copy the `codacy-seed-0.0.1.tgz` inside the docker you will be building

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
    * Add `!codacy-seed-0.0.1.tgz` to your `.dockerignore`
```

>     *  Add `!codacy-seed-0.0.1.tgz` to your `.dockerignore`

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

     *  Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`
```

>     *  Add `!codacy-seed-0.0.1.tgz` to your `.dockerignore`

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
     -  Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`
```

>      *  Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`

Author: Codacy has a fix for the issue: [Expected: 0; Actual: 3](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
*  Remove multi-stage docker steps
```

>      *  Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
   -  Remove multi-stage docker steps
```

>      *  Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
   -  Remove multi-stage docker steps
```

>      *  Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
   * Remove multi-stage docker steps
```

>      *  Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`

Author: Codacy has a fix for the issue: [Expected: 0; Actual: 3](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
*  Remove multi-stage docker steps
```

>    *  Remove multi-stage docker steps

Author: Codacy has a fix for the issue: [Expected: 2; Actual: 8](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
  *  Lines from `FROM node:$NODE_IMAGE_VERSION` to `RUN rm -rf /package.json /package-lock.json`
```

>    *  Remove multi-stage docker steps

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
        * Lines from `FROM node:$NODE_IMAGE_VERSION` to `RUN rm -rf /package.json /package-lock.json`
```

>    *  Remove multi-stage docker steps

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
        -  Lines from `FROM node:$NODE_IMAGE_VERSION` to `RUN rm -rf /package.json /package-lock.json`
```

>    *  Remove multi-stage docker steps

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
        -  Lines from `FROM node:$NODE_IMAGE_VERSION` to `RUN rm -rf /package.json /package-lock.json`
```

>    *  Remove multi-stage docker steps

Author: Codacy has a fix for the issue: [Expected: 2; Actual: 8](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
  *  Lines from `FROM node:$NODE_IMAGE_VERSION` to `RUN rm -rf /package.json /package-lock.json`
```

        *  Lines from `FROM node:$NODE_IMAGE_VERSION` to `RUN rm -rf /package.json /package-lock.json`
        > This way you skip copying the files to the other docker, and another `npm install`

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
5. Publish your docker locally as normal: `docker build -t codacy-eslint:local .`
```

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
5. Publish your docker locally as normal: `docker build -t codacy-eslint:local .`
```

# Codacy ESLint

## Adding new plugins / configs

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
1. Install the package using npm:
```

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
1. Install the package using npm:
```

1.  Install the package using npm:

    ```bash
    npm install <package-name>
    ```

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
2. \[Plugins only\] Add the plugin to the plugins section in the file `src/eslintDefaultOptions.ts`
```

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
2. \[Plugins only\] Add the plugin to the plugins section in the file `src/eslintDefaultOptions.ts`
```

2.  \[Plugins only\] Add the plugin to the plugins section in the file `src/eslintDefaultOptions.ts`

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
3. \[Plugins only\] If the plugin has descriptions for rules on GitHub, reference them
```

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
3. \[Plugins only\] If the plugin has descriptions for rules on GitHub, reference them
```

3.  \[Plugins only\] If the plugin has descriptions for rules on GitHub, reference them
at `src/docGeneratorMain.ts` to include them on the generated documentation. To do this, add a section similar to the following example:

    ```typescript
    console.log("Generate xss description files")
    await docGenerator.downloadDocs(
      (pattern) =>
        `${githubBaseUrl}/Rantanen/eslint-plugin-xss/master/docs/rules/${pattern}.md`,
      "xss"
    )
    ```

4.  Generate documentation so it adds the new plugin documentation

## Generating documentation

```bash
npm install
npm run generateDocs
```

> ## Test changes to codacy-seed locally

Author: Codacy has a fix for the issue: [(codacy-engine-typescript-seed)[https://github.com/codacy/codacy-engine-typescript-seed]](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
You may need to test changes that comes from our [codacy-engine-typescript-seed](https://github.com/codacy/codacy-engine-typescript-seed).
```

You may need to test changes that comes from our (codacy-engine-typescript-seed)[https://github.com/codacy/codacy-engine-typescript-seed].

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
1. Create a package with your changes on the seed:
```

1.  Create a package with your changes on the seed:
    - Don't forget to update the dependencies: `npm install`
    - Compile the library: `npm run compile`
    - Package the library: `npm pack`
        > This should generate a codacy-seed-0.0.1.tgz on your codacy-seed repository

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
2. Copy the `codacy-seed-0.0.1.tgz` into the root of this repository
```

2.  Copy the `codacy-seed-0.0.1.tgz` into the root of this repository

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
3. Install the package: `npm install codacy-seed-0.0.1.tgz`
```

3.  Install the package: `npm install codacy-seed-0.0.1.tgz`

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
4. Update Dockerfile and `.dockerignore` so you copy the `codacy-seed-0.0.1.tgz` inside the docker you will be building
```

> 4.  Update Dockerfile and `.dockerignore` so you copy the `codacy-seed-0.0.1.tgz` inside the docker you will be building

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
    -  Add `!codacy-seed-0.0.1.tgz` to your `.dockerignore`
```

>     *  Add `!codacy-seed-0.0.1.tgz` to your `.dockerignore`

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
     -  Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`
```

>     *  Add `!codacy-seed-0.0.1.tgz` to your `.dockerignore`

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
     * Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`
```

>     *  Add `!codacy-seed-0.0.1.tgz` to your `.dockerignore`

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
     -  Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`
```

>      *  Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
   * Remove multi-stage docker steps
```

>      *  Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`

Author: Codacy has a fix for the issue: [Expected: 0; Actual: 3](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
*  Remove multi-stage docker steps
```

>      *  Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
   -  Remove multi-stage docker steps
```

>      *  Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

   *  Remove multi-stage docker steps
```

>      *  Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`

Author: Codacy has a fix for the issue: [Expected: 0; Actual: 3](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
*  Remove multi-stage docker steps
```

>      *  Add the package to the docker before `RUN npm install`: `COPY codacy-seed-0.0.1.tgz ./`

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
   -  Remove multi-stage docker steps
```

>    *  Remove multi-stage docker steps

Author: Codacy has a fix for the issue: [Expected: dash; Actual: asterisk](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
        -  Lines from `FROM node:$NODE_IMAGE_VERSION` to `RUN rm -rf /package.json /package-lock.json`
```

>    *  Remove multi-stage docker steps

Author: Codacy has a fix for the issue: [Expected: 2; Actual: 8](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
  *  Lines from `FROM node:$NODE_IMAGE_VERSION` to `RUN rm -rf /package.json /package-lock.json`
```

>    *  Remove multi-stage docker steps

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
        * Lines from `FROM node:$NODE_IMAGE_VERSION` to `RUN rm -rf /package.json /package-lock.json`
```

>    *  Remove multi-stage docker steps

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
        * Lines from `FROM node:$NODE_IMAGE_VERSION` to `RUN rm -rf /package.json /package-lock.json`
```

        *  Lines from `FROM node:$NODE_IMAGE_VERSION` to `RUN rm -rf /package.json /package-lock.json`
        > This way you skip copying the files to the other docker, and another `npm install`

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
5. Publish your docker locally as normal: `docker build -t codacy-eslint:local .`
```

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 2](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion
5. Publish your docker locally as normal: `docker build -t codacy-eslint:local .`
```

5.  Publish your docker locally as normal: `docker build -t codacy-eslint:local .`

## Limitations

### Incompatible rules

There are some ESLint rules that will be ignored when running this Docker container. For more details on the ignored
rules, check `blacklistRegexes` defined at [blacklist.ts](src/blacklist.ts).

Usually, these rules need an Internet connection and/or to check `node_modules`, and would not run successfully
on our Docker container environment.

## What is Codacy

[Codacy](https://www.codacy.com/) is an Automated Code Review Tool that monitors your technical debt, helps you improve your code quality, teaches best practices to your developers, and helps you save time in Code Reviews.

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 0; Below](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

### Among Codacy’s features
```

> 

Author: Codacy has a fix for the issue: [Expected: 1; Actual: 0; Below](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

### Among Codacy’s features
```

> ### Among Codacy’s features

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Identify new Static Analysis issues
```

> ### Among Codacy’s features

Author: Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/franciscodua/codacy-test/pullRequest?prid=7573067)
```suggestion

- Identify new Static Analysis issues
```
